### PR TITLE
[memstore-plugin]Replace ot with otel sdk

### PR DESCRIPTION
**Pull request by afzal442**
## Which problem is this PR solving?
* Part of #3381 
* This PR adds `otel` traces and removes deprecated `jaeger-client-go` api

## Short description of the changes
- Replace OT tracer with OTEL

### CODE REVIEW

File: examples/memstore-plugin/main.go [hunk 1]
```diff
@@ -55,21 +53,11 @@ func main() {
...
-			Param: 1.0,
-		},
-		RPCMetrics: true,
-	}
-
-	tracer, closer, err := traceCfg.NewTracer()
+	tracer, err := jtracer.New("mem-store")
 	if err != nil {
-		panic("Failed to initialize tracer")
+		log.Fatal("Failed to initialize tracer", zap.Error(err))
```
Comment (yurishkuro) [inline examples/memstore-plugin/main.go:58]: zap.Error() is from another logging library.
```suggestion
		panic(fmt.Errorf("Failed to initialize tracer: %w", err))
```

File: examples/memstore-plugin/main.go [hunk 2]
```diff
@@ -16,25 +16,22 @@ package main
...
 
+	"github.com/jaegertracing/jaeger/pkg/jtracer"
 	"github.com/jaegertracing/jaeger/plugin/storage/grpc"
 	grpcMemory "github.com/jaegertracing/jaeger/plugin/storage/grpc/memory"
 	"github.com/jaegertracing/jaeger/plugin/storage/grpc/shared"
 	"github.com/jaegertracing/jaeger/plugin/storage/memory"
 )
 
-const (
-	serviceName = "mem-store"
```
Comment (yurishkuro) [inline examples/memstore-plugin/main.go:35]: you didn't have to delete this, the string is still used

File: examples/memstore-plugin/main.go [hunk 3]
```diff
@@ -55,21 +52,11 @@ func main() {
...
-		},
-		RPCMetrics: true,
-	}
-
-	tracer, closer, err := traceCfg.NewTracer()
+	tracer, err := jtracer.New("mem-store")
 	if err != nil {
-		panic("Failed to initialize tracer")
+		panic(fmt.Errorf("failed to initialize tracer: %w", err))
 	}
```
Comment (yurishkuro) [inline examples/memstore-plugin/main.go:58]:
```suggestion
	}
	defer tracer.Close()
```

Comment (afzalbin64) [reply examples/memstore-plugin/main.go:58]: wow, how I can forget that. Thanks Yuri

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**